components/camel-sjms/src/test/java/org/apache/camel/component/sjms/consumer/AsyncConsumerFalseTest.java

```diff
@@ -58,17 +58,13 @@ public void testAsyncJmsConsumer() throws Exception {
     }
 
 
-    protected CamelContext createCamelContext() {
-
-
+    @Override
+    protected void configureCamelContext(CamelContext camelContext) {
         ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory(
                 service.serviceAddress());
         SjmsComponent component = new SjmsComponent();
         component.setConnectionFactory(connectionFactory);
-        context.addComponent("sjms", component);
-
-
-        return context;
+        camelContext.addComponent("sjms", component);
```

Comment:
Although this is shown on the top, this is actually the 3rd important item (read the other 2 before going back to this). 

This is something we won't be able to get rid of. The original method was relying on two things that cannot be true anymore:

1. That the parent class would create the context (i.e.: it has something like `super.createContext()` but that has to be removed. 
2. And it was also not being called by the parent class.